src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -0,0 +1,392 @@
+     * offsets.
+     * Note: This internal-use constructor keeps an alias pointer to the given list.
+     */
+    @VisibleForTesting
+    ReadPileup(final Locatable loc, final List<PileupElement> pileup) {
+        this.loc = loc;
+        this.pileupElements = pileup;
+    }
+
+    /**
+     * Create a new pileup without any aligned reads
+     */
+    public ReadPileup(final Locatable loc) {
+        this(loc, Collections.emptyList());
+    }
+
+    /**
+     * Create a new pileup with the given reads.
+     */
+    public ReadPileup(final Locatable loc, final List<GATKRead> reads, final int offset) {
+        this(loc, readsOffsets2Pileup(reads, offset));
+    }
+
+
+    /**
+     * Create a new pileup with the given reads.
+     */
+    public ReadPileup(final Locatable loc, final List<GATKRead> reads, final List<Integer> offsets) {
+        this(loc, readsOffsets2Pileup(reads, offsets));
+    }
+
+    /**
+     * Helper routine for converting reads and offset lists to a PileupElement list.
+     */
+    private static List<PileupElement> readsOffsets2Pileup(final List<GATKRead> reads, final List<Integer> offsets) {
+        if (reads == null) {
+            throw new GATKException("Illegal null read list in UnifiedReadBackedPileup");
+        }
+        if (offsets == null) {
+            throw new GATKException("Illegal null offsets list in UnifiedReadBackedPileup");
+        }
+        if (reads.size() != offsets.size()) {
+            throw new GATKException("Reads and offset lists have different sizes!");
+        }
+
+        final List<PileupElement> pileup = new LinkedList<>();
+        for (int i = 0; i < reads.size(); i++) {
+            pileup.add(PileupElement.createPileupForReadAndOffset(reads.get(i), offsets.get(i)));
+        }
+
+        return pileup;
+    }
+
+    /**
+     * Helper routine for converting reads and a single offset to a PileupElement list.
+     */
+    private static List<PileupElement> readsOffsets2Pileup(final List<GATKRead> reads, final int offset) {
+        if (reads == null) {
+            throw new GATKException("Illegal null read list in UnifiedReadBackedPileup");
+        }
+        if (offset < 0) {
+            throw new GATKException("Illegal offset < 0 UnifiedReadBackedPileup");
+        }
+
+        final List<PileupElement> pileup = new LinkedList<>();
+        for (final GATKRead read : reads) {
+            pileup.add(PileupElement.createPileupForReadAndOffset(read, offset));
+        }
+
+        return pileup;
+    }
+
+    /**
+     * Make a new pileup consisting of elements of this pileup that satisfy the predicate.
+     */
+    public ReadPileup makeFilteredPileup(final Predicate<PileupElement> filter){
+        return new ReadPileup(loc, pileupElements.stream().filter(filter).collect(Collectors.toList()));
+    }
+
+    /**
+     * Make a new pileup consisting of only reads on the negative strand.
```

Comment:
Use forward/reverse terminology rather than positive/negative (here and below) for consistency with the `GATKRead` interface.